File: frontend/src/types/validation.ts
export type ValidationState = 'PASS' | 'FAIL' | 'WARNING' | 'NOT_TESTED';

export interface ValidationCheck {
  id: string;
  status: ValidationState;
  detail: string | null;
}

export interface ValidationReport {
  run_id: string | null;
  scenario: string;
  status: ValidationState;
  validation: Record<string, ValidationState>;
  checks: ValidationCheck[];
  checked_at: string | null;
}

File: backend/app/models/validation.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

ValidationState = Literal["PASS", "FAIL", "WARNING", "NOT_TESTED"]


class ValidationCheck(BaseModel):
    id: str
    status: ValidationState
    detail: str | None = None


class ValidationReport(BaseModel):
    run_id: str | None = None
    scenario: str
    status: ValidationState
    validation: dict[str, ValidationState]
    checks: list[ValidationCheck] = Field(default_factory=list)
    checked_at: datetime | None = None
